### PR TITLE
[5.1] Allow validateIn for type array

### PR DESCRIPTION
**Pull request by dontbug**
Allow for validation:

``` php
$inputs = [
    'values' => ['one']
];
$rules = [
    'values' => 'required|in:one,two'
];
$v = Validator::make($inputs, $rules);
var_dump(!$v->fails());
```

### CODE REVIEW

File: src/Illuminate/Validation/Validator.php [hunk 1]
```diff
@@ -962,6 +962,10 @@ protected function getSize($attribute, $value)
      */
     protected function validateIn($attribute, $value, $parameters)
     {
+        if (in_array('array', $this->rules[$attribute]) && is_array($value)) {
+            return (count(array_diff($value, $parameters)) == 0);
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: Don't need the extra brackets here. :)

File: src/Illuminate/Validation/Validator.php [hunk 2]
```diff
@@ -962,6 +962,10 @@ protected function getSize($attribute, $value)
      */
     protected function validateIn($attribute, $value, $parameters)
     {
+        if (in_array('array', $this->rules[$attribute]) && is_array($value)) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: Moving the is_array check first would be faster.

```php
if (is_array($value) && in_array('array', $this->rules[$attribute])) {
```

**Comment by Talean-dev** [comment]
executed. Sorry :)

**Comment by Talean-dev** [comment]
i'am sorry :smile:

**Comment by GrahamCampbell** [comment]
Please squash to one commit, then this looks good. :)

**Comment by taylorotwell** [comment]
What if the array has more than one value?

**Comment by Talean-dev** [comment]
```php
 if (is_array($value) && in_array('array', $this->rules[$attribute])) {
            return count(array_diff($value, $parameters)) == 0;
        }
```
Check if all the values array to be in the rules

**Comment by taylorotwell** [comment]
Can you add to the unit tests perhaps?

**Comment by Talean-dev** [comment]
Example used:
Example

Yes, tomorrow will be able to focus on creating Unit tests.

**Pull request merged**